includes/payment-methods/Configs/Definitions/LinkDefinition.php

```diff
@@ -0,0 +1,200 @@
+<?php
+/**
+ * Link Payment Method Definition
+ *
+ * @package WCPay\PaymentMethods\Configs\Definitions
+ */
+
+namespace WCPay\PaymentMethods\Configs\Definitions;
+
+use WCPay\PaymentMethods\Configs\Interfaces\PaymentMethodDefinitionInterface;
+use WCPay\PaymentMethods\Configs\Constants\PaymentMethodCapability;
+use WCPay\Constants\Currency_Code;
+use WCPay\PaymentMethods\Configs\Utils\PaymentMethodUtils;
+
+/**
+ * Class implementing the Link payment method definition.
+ */
+class LinkDefinition implements PaymentMethodDefinitionInterface {
+
+	/**
+	 * Get the internal ID for the payment method
+	 *
+	 * @return string
+	 */
+	public static function get_id(): string {
+		return 'link';
+	}
+
+	/**
+	 * Get the keywords for the payment method. These are used by the duplicate detection service.
+	 *
+	 * @return string[]
+	 */
+	public static function get_keywords(): array {
+		return [ 'link', 'stripe link' ];
+	}
+
+	/**
+	 * Get the Stripe payment method ID
+	 *
+	 * @return string
+	 */
+	public static function get_stripe_id(): string {
+		return PaymentMethodUtils::get_stripe_id( self::get_id() );
+	}
+
+	/**
+	 * Get the customer-facing title of the payment method
+	 *
+	 * @param string|null $account_country Optional. The merchant's account country.
+	 *
+	 * @return string
+	 */
+	public static function get_title( ?string $account_country = null ): string {
+		return __( 'Link', 'woocommerce-payments' );
+	}
+
+	/**
+	 * Get the title of the payment method for the settings page.
+	 *
+	 * @param string|null $account_country Optional. The merchant's account country.
+	 *
+	 * @return string
+	 */
+	public static function get_settings_label( ?string $account_country = null ): string {
+		return self::get_title( $account_country );
+	}
+
+	/**
+	 * Get the customer-facing description of the payment method
+	 *
+	 * @param string|null $account_country Optional. The merchant's account country.
+	 *
+	 * @return string
+	 */
+	public static function get_description( ?string $account_country = null ): string {
+		return __( 'Link autofills your customers\' payment and shipping details to deliver an easy and seamless checkout experience.', 'woocommerce-payments' );
+	}
+
+	/**
+	 * Get the list of supported currencies
+	 *
+	 * @return string[] Array of currency codes
+	 */
+	public static function get_supported_currencies(): array {
+		return [];
+	}
+
+	/**
+	 * Get the list of supported countries
+	 *
+	 * @param string|null $account_country Optional. The merchant's account country.
+	 *
+	 * @return string[] Array of country codes
+	 */
+	public static function get_supported_countries( ?string $account_country = null ): array {
+		return [];
+	}
```

Comment:
As discussed in WOOPMNT-5518, we need to decide when to enable Link. I see that simply setting this to `[ CountryCode::US ]` is not sufficient, but I think we should make sure not to let Link be widely available by merging this PR.

Reply:
That's fair. In order to not introduce new behavioral changes, I re-introduced the currency code limitation for Stripe Link in b2b4caa7b , same as the old `includes/payment-methods/class-link-payment-method.php`

Stripe Link is already limited to US-based Stripe Accounts via server-side. The `get_supported_countries` method in the payment method definitions is used during checkout for _customer_ countries.

I'll fix WOOPMNT-5518 in a separate PR.